src/main/java/racingcar/domain/car/Car.java

```diff
@@ -0,0 +1,38 @@
+package racingcar.domain.car;
+
+import racingcar.strategy.MoveStrategy;
+
+public class Car {
+    private final Name name;
+    private final Position position;
+
+    public Car(String name) {
+        this.name = new Name(name);
+        this.position = new Position(0);
+    }
+
+    private Car(Name name, int position) {
+        this.name = name;
+        this.position = new Position(position);
+    }
+
+    public Name getName() {
+        return name;
+    }
+
+    public Position getPosition() {
+        return position;
+    }
+
+    public Car move(MoveStrategy moveStrategy) {
+        int nextPosition = position.getPosition();
+        if (moveStrategy.isMove()) {
+            nextPosition++;
+        }
+        return new Car(name, nextPosition);
+    }
+
+    public boolean isWinnerPosition(int winnerPosition) {
+        return position.getPosition() == winnerPosition;
+    }
```

Comment:
getPosition().getPosition()이 뭔가 어색하지 않나요? 저는 이런식으로 구현했을것 같아요.

```suggestion
    public boolean isPositionEquals(Position position) {
        return this.position.equals(position);
    }
```

```java
public class Cars {

    ...

    public List<String> getMaxPositionCarNames() {
        Position max = cars.stream()
                .max(Comparator.comparing(Car::getPosition))
                .orElseThrow(() -> new IllegalStateException("cannot find, maybe no cars"))
                .getPosition();

        return cars.stream()
                .filter(car -> car.isPositionEquals(max))
                .map(car -> car.getName().getName())
                .collect(Collectors.toList());
    }
```

원시값을 포장한 값 객체(VO)는 자신이 만든 소프트웨어 세상의 소통 도구입니다. 불변 객체로 만들면 안전하니, 마음것 사용할 수도 있구요.

마찬가지로 `car.getName().getName()` 역시 `Winner` 가 `List<Name> winnerNames;` 를 가졌다면 안적혔을것 같아요 🙂